apps/designer/app/designer/features/props-panel/control.tsx

```diff
@@ -142,84 +220,150 @@ const RangeControl = ({
     </Flex>
   </Flex>
 );
+*/
 
 const NotImplemented = () => <div />;
 
-type PrimitiveControlProps = BaseControlProps & {
-  type: "array" | "boolean" | "date" | "number" | "object" | "text";
+const includes = <T extends string>(arr: readonly T[], v: string): v is T => {
+  return arr.includes(v as never);
 };
 
-type ColorControlProps = BaseControlProps & {
-  type: "color";
-  presetColors?: Array<string>;
+const isStringArray = (arr: unknown): arr is Array<string> => {
```

Comment:
this could be also a good utility in the same belt

Reply:
Disagree, people even don't reuse npm packages already exists in big repos, and reusing a one-liner like this is nearly impossible. I think I wrote string array type guard for the first time in my life, and the same for includes. They are rare, the main reason is to be type guards, and not every piece of code should has own module or be in some utils. Even so - reusability is near impossible.

Reply:
yeah maybe this one is a bit too special